internal/controllers/clusterextension_controller.go

```diff
@@ -390,12 +390,18 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 			_, isWatched := r.dynamicWatchGVKs[unstructuredObj.GroupVersionKind()]
 			if !isWatched {
 				if err := r.controller.Watch(
-					source.Kind(r.cache, unstructuredObj),
-					crhandler.EnqueueRequestForOwner(r.Scheme(), r.RESTMapper(), ext, crhandler.OnlyControllerOwner()),
-					helmpredicate.DependentPredicateFuncs()); err != nil {
+					source.Kind(
+						r.cache, unstructuredObj, crhandler.TypedEnqueueRequestForOwner[*unstructured.Unstructured](
+							r.Scheme(),
+							r.RESTMapper(),
+							ext,
+							crhandler.OnlyControllerOwner(),
+						),
+						helmpredicate.DependentPredicateFuncs[*unstructured.Unstructured](),
+					),
+				); err != nil {
```

Comment:
I have a slightly different version of this in #912:
https://github.com/operator-framework/operator-controller/pull/912/files#diff-a54f79b762054ccdc31e4011d28cce22af8f54d43e1fddc75745ce042f177f7e

Reply:
To reduce the noise in this PR, I'm going to suggest the k8s upgrade happen in #912. But, there's a bit of different code in how this section is fixed up, and I'm not sure which one is better.

Reply:
The difference in #912 is that you're setting the watches on objects directly, whereas this one is first modifying to an unstructured type. One benefit of using unstructured is being able to work with dynamic clients. I don't know the exact reason on why it was done this way, but the change to unstructured preserves the previous functionality.

Reply:
I wrote this originally I think. My memory is fuzzy, but I suspect unstructured was used because we need to support arbitrary objects. If a structured type is used, it needs to be added to the scheme. But we can't know how to add arbitrary types to a scheme. Hence, the use of unstructured.

This code has been around for like 5 years though, so circumstances and assumptions may have changed.

@tmshort I think the best way to test your change would be to craft a bundle that has an arbitrary CR in it, install that bundle, and then see if this dynamic watch setup is successful.

Reply:
During install, any resource present in the bundle has the owner refs set to the cluster extension.

Reply:
It appears I can't attempt to use this version, because it requires an update to`helmpredicate`, which is in rukpak.

Reply:
I would prefer we keep the `main` version of this code, tweaked as necessary.